<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <arg name="robot" default="pr2"/>
  <!-- Camera parameters -->
  <arg name="rgb_image_topic" default="kinect2/qhd/image_color_rect"/>
  <arg name="rgb_camera_info_topic" default="kinect2/qhd/camera_info"/>
  <arg name="depth_image_topic" default="kinect2/qhd/image_depth_rect"/>
  <arg name="depth_camera_info_topic" default="kinect2/qhd/camera_info"/>

  <arg name="global_frame_id" default="map"/>
  <arg name="base_frame_id" default="base_footprint"/>

  <!-- Simulation parameters -->
  <arg name="use_simulation_gui" default="true"/>
  <arg name="simulation_config_filename" default="$(find uwds3)/config/simulation/pr2_config.yaml"/>
  <arg name="cad_models_additional_search_path" default="$(find uwds3)/models/cad_models"/>
  <!-- <arg name="static_entities_config_filename" default="$(find uwds3)/config/simulation/empty_floor.yaml"/> -->

  <arg name="static_entities_config_filename" default="$(find uwds3)/config/simulation/tabletop_setup.yaml"/>
  <arg name="robot_urdf_file_path" default="urdf/$(arg robot).urdf"/>

  <!-- Other parameters -->
  <arg name="publish_viz" default="true"/>
  <arg name="publish_tf" default="true"/>
  <arg name="use_depth" default="false"/>

  <node name="internal_simulator" pkg="uwds3" type="internal_simulator_node.py" args="" output="screen" respawn="false">
    <rosparam subst_value="true">

      rgb_image_topic: $(arg rgb_image_topic)
      rgb_camera_info_topic: $(arg rgb_camera_info_topic)

      depth_image_topic: $(arg depth_image_topic)
      depth_camera_info_topic: $(arg depth_camera_info_topic)

      global_frame_id: $(arg global_frame_id)
      base_frame_id: $(arg base_frame_id)

      use_simulation_gui: $(arg use_simulation_gui)
      simulation_config_filename: $(arg simulation_config_filename)
      cad_models_additional_search_path: $(arg cad_models_additional_search_path)
      static_entities_config_filename: $(arg static_entities_config_filename)
      robot_urdf_file_path: $(arg robot_urdf_file_path)

      publish_viz: $(arg publish_viz)
      publish_tf: $(arg publish_tf)

      use_depth: $(arg use_depth)

    </rosparam>
  </node>

</launch>
